presto-main/src/main/java/com/facebook/presto/execution/RenameColumnTask.java

```
@@ -52,7 +53,8 @@ public String getName()
         Session session = stateMachine.getSession();
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTable());
         Optional<TableHandle> tableHandleOptional = metadata.getTableHandle(session, tableName);
-        if (!tableHandleOptional.isPresent()) {
+        Optional<ConnectorMaterializedViewDefinition> optionalMaterializedView = metadata.getMaterializedView(session, tableName);
+        if (!tableHandleOptional.isPresent() || optionalMaterializedView.isPresent()) {
```

same, we should throw errors other than MISSING_TABLE